src/index.js

```diff
@@ -0,0 +1,164 @@
+'use strict';
+// Leave this commented out:
+// const { default: axios } = require("axios");
+
+const state = {
+  temperature: '',
+  cityName: 'New York City',
+  incButton: null,
+  decButton: null,
+  skyDropdown: null,
+  resetButton: null,
+  realtimeButton: null,
+};
+
+const makeRealtimeTempUpdates = () => {
+  // calls proxy API to get lat and lon
+  getLatLon(state.cityName)
+    .then(({ latitude, longitude }) => {
+      // calls proxy API to get temp
+      getWeather(latitude, longitude)
+        .then(({ tempK }) => {
+          // update temp state
+          updateTempState(tempK);
+          updateTempOnWebpage();
+          changeColorAndLandscape();
+        })
+        .catch((error) => {
+          console.log('error in getWeather!');
+        });
+    })
+    .catch((error) => {
+      console.log('error in getLatLon!');
+    });
+};
+
+const getLatLon = async (city) => {
+  const latLonResponse = await axios.get('http://127.0.0.1:5000/location', {
+    params: {
+      q: city,
+    },
+  });
+  const latitude = latLonResponse.data[0].lat;
+  const longitude = latLonResponse.data[0].lon;
+  return { latitude, longitude };
+};
+
+const getWeather = async (apiLatitude, apiLongitude) => {
+  const weatherResponse = await axios.get('http://127.0.0.1:5000/weather', {
+    params: {
+      lat: apiLatitude,
+      lon: apiLongitude,
+    },
+  });
+  const tempK = weatherResponse.data.main.temp;
+  return { tempK };
+};
+
+const updateTempState = (tempInKelvin) => {
+  // converts temp to Farenheit and saves it to the state
+  state.temperature = Math.floor(((tempInKelvin - 273.15) * 9) / 5 + 32);
+};
+
+const increaseTemp = () => {
+  state.temperature += 1;
+  updateTempOnWebpage();
+  changeColorAndLandscape();
+};
+
+const decreaseTemp = () => {
+  state.temperature -= 1;
+  updateTempOnWebpage();
+  changeColorAndLandscape();
+};
+
+const updateTempOnWebpage = () => {
+  const websiteTemp = document.getElementById('temp');
+  websiteTemp.textContent = `${state.temperature}°F`;
+};
+
+const changeColorAndLandscape = () => {
```

Comment:
How might you refactor this so the logic is more data driven? Here you have several if/else if statements. What if you had an object with the keys as the type of weather and the values as the corresponding display? Same with the temperature values! Then you could iterate over the object to set `websiteTemp` and `websiteLandscape`